calculator-project/src/components/buttons/buttons.js

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import "../buttons/buttons.css";
+import { create, all } from "mathjs";
+// import numeral from "numeral";
+
+const config = { };
+const math = create(all, config);
+
+class Buttons extends React.Component{
+    state= {
+        NumberButtons: [{oName: "zero", value: "0"}, {oName: "one", value: "1"}, {oName: "two", value: "2"}, {oName: "three", value: "3"}, {oName: "four", value: "4"}, {oName: "five", value: "5"}, {oName: "six", value: "6"}, {oName: "seven", value: "7"}, {oName: "eight", value: "8"}, {oName: "nine", value: "9"}],
+        OperatorButtons: [{oName: "decimal", value: "."},{ oName: 'add', value: "+"}, { oName: 'subtract', value: "-"}, { oName: 'multiply', value: "*"}, { oName: 'divide', value: "/"}, { oName: 'polar', value: "±"}, {oName: 'percentage', value: "%"}, { oName: 'clear', value: "C"}, { oName: 'equal', value: "="}],
+        displayScreen: ""
+    }
+    
+
+    populateButtons = (arr) => {
+        return arr.map((button, index) =>{
+            const { oName, value } = button
+            if(value === "=" || value === "%" || value === "C" || value === "±") {
+                return <button key={index} className="buttons" id={oName} value={value} onClick={this.calculate}>{value}</button>
+            } else {
+                return <button key={index} className="buttons" id={oName} value={value} onClick={this.display}>{value}</button>
+            }
+        })
+    }
+
+    display = (e) => {
+        e.preventDefault();
+        let value = e.target.value
+        let updated = this.state.displayScreen;
+        updated += value;
+        // let commas = this.numberWithCommas(updated)
+        this.setState({displayScreen:updated});
+    }
+
+//    numberWithCommas =(x)=> {
+//         return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+//     }
+
+calculate = (e) => {
+    e.preventDefault();
```

Comment:
We usually use `e.preventDefault()` on forms , because users might want to use the Enter button on the keyboard instead of the submit button on the webpage. However on the calculator that's unnecessary.